core/src/main/java/org/mule/runtime/core/internal/routing/IdempotentMessageValidator.java

```diff
@@ -72,11 +76,31 @@ public void initialise() throws InitialisationException {
           format("%s.%s.%s", muleContext.getConfiguration().getId(), getLocation().getRootContainerName(),
                  this.getClass().getName());
     }
+    setupObjectStore();
+  }
+
+  private void setupObjectStore() throws InitialisationException {
+    //Check if  OS was properly configured
+    if (store != null && privateStore != null) {
+      throw new InitialisationException(createStaticMessage("Ambiguous definition of object store, both reference and private were configured"),
+                                        this);
+    }
     if (store == null) {
-      this.store = createMessageIdStore();
+      if (privateStore == null) //If no object store was defined, create one
+      {
+        this.store = createMessageIdStore();
+      } else { //If object store was defined privatelly
+        this.store = privateStore;
+      }
+    }
+    initialiseIfNeeded(store, true, muleContext);
```

Comment:
yes. Why do you speculate with Object stores being initialisable and startable but not the other two? It's very common that if you implement any of those two, then you have at least one of the other to do clean up